### DISCUSSION TITLE
Unable to compile in Ardunio  Correctly and always get a 1 in the TFT didplay when fully started

### DISCUSSION THREAD

**Discussion by sivaelid** [dl9rdz/rdz_ttgo_sonde#404] [Q&A]
I am  using the TTGO v1.2 with AXP2101. I know not fully suported  for me the issue is that it does not enable charging of the battery. I know what i need to do to enable this. I though cannot complie the code in a working stated to start with while If I follow the instruction to manualy compile in Ardunio, I get two issue.

1. The Most upto date GFX library generates  lots of errors I have go back from 1.4 ( current one) to 1.2 and that compiles with out error .
2. THAt one compiled  it says It can not find networks even though The Network are in the networks list , but also once started the display show welcome display and clears and the number 1 in then displayed and oneing else works  ? 

Any one compiled form scratch in Ardunio and not using the Bin Files. Would welcome some hel to get a good compiled verson the works.

Regards Simon

**Comment by dl9rdz** [comment]
I now have obtained a TTGO v1.2 with AXP2101 myself and will do some testing probably next weekend.

You can look at what the automated build of the binary images does (it uses Arduino to compile) in the file .travis.yml

It uses the library GFX Library for Arduino version 1.1.5

You can also use platform.io to build images. These are slightly larger than the Arduino build but if you don't indent to add much code it should still fit into the standard image size.  The platform.ini contains all references to the versions of library that work, so it should be as simple as "pio run" to build the image on the command line, or use the platformIO IDE.

**Comment by sivaelid** [comment]
Hi di9rdz  Thanks for the update above  Yes still getting compile issue Not sure what doing wrong. I will wait to after  you ahve had a look at it. The mood tHat I wanted to do are in the  issues namly get the unit to charge a battery in the T-Beam. The register files need to be set to initiatiate the charge  as the default seems to be  set to off regardless to the data sheet. I tried the Xpower Libray example to see if it charged the board and that works correctly with that example . In that case set the Cut off voltage to 4.2 v and 550mA constant charge..  Thanks   Simon G8HAM
